src/fileviewer.cpp

```diff
@@ -392,14 +394,41 @@ void FileViewer::ShowError(const char *icon, const wxString& msg, const wxString
 
 void FileViewer::OnChoice(wxCommandEvent &event)
 {
-    SelectReference(m_references[event.GetSelection()]);
+    m_chosenRefIndex = event.GetSelection();
+    SelectReference(m_references[m_chosenRefIndex]);
 }
 
-void FileViewer::OnEditFile(wxCommandEvent&)
+void FileViewer::OnEditFile(wxCommandEvent &)
 {
     wxFileName filename = GetFilename(bidi::strip_control_chars(m_file->GetStringSelection()));
-    if (filename.IsOk())
-        wxLaunchDefaultApplication(filename.GetFullPath());
+    if (filename.IsOk()) {
+        wxFileType *c_type;
+        c_type = wxTheMimeTypesManager->GetFileTypeFromExtension(filename.GetExt());
+        if (c_type) {
+    	      wxString command, filePath;
+    	      filePath = filename.GetFullPath();
+    	      wxFileType::MessageParameters param;
+    	      wxFileType::MessageParameters &paramRef = param;
+    	      c_type->GetOpenCommand(&command, paramRef);
```

Comment:
This is a rather non-standard use of `GetOpenCommand`, relying on the filename being empty. I understand why you did it, but it could use an explicit comment...

Reply:
As you know to extract the command name alone, it is a way. I added more detailed comment.